internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -462,6 +462,18 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, log logrus.FieldLogge
 	log.Debugf("Started BMH reconcile for %s/%s", bmh.Namespace, bmh.Name)
 	log.Debugf("BMH value %v", bmh)
 
+	// A detached BMH is considered to be unmanaged by the hub
+	// cluster and, therefore, BMAC reconciles on this BMH should
+	// not happen.
+	//
+	// User is expected to remove the `detached` annotation manually
+	// to bring this BMH back into the pool of reconciled BMH resources.
+	bmhAnnotations := bmh.ObjectMeta.GetAnnotations()
+	if _, ok := bmhAnnotations[BMH_DETACHED_ANNOTATION]; ok {
```

Comment:
I'm not sure if my way of thinking is correct, but what if someone (for any reason) set `baremetalhost.metal3.io/detached: false` ? Wouldn't it be better&safer if we were explicitly checking for `bmh.ObjectMeta.Annotations[BMH_DETACHED_ANNOTATION] == "true"` ?

Reply:
Your thinking is correct. However, in this case, this code follows what BMO does:

https://github.com/metal3-io/baremetal-operator/blob/f5228fa38953b0e0e3f4763797013a3f099151dc/controllers/metal3.io/host_state_machine.go#L242-L250

Mostly trying to be consistent with that (wish this function would be exported, tbh)

Reply:
As discussed internally, what you do is correct